src/subcommand/minimizer_main.cpp

```diff
@@ -377,61 +377,65 @@ int main_minimizer(int argc, char** argv) {
         std::cerr << std::endl;
     }
     if (distance_name.empty()) {
-        gbwtgraph::index_haplotypes(gbz->graph, index, [](const pos_t&) -> gbwtgraph::Payload {
-            return MIPayload::NO_CODE;
-        });
-    } else {
-        gbwtgraph::index_haplotypes(gbz->graph, index, [&](const pos_t& pos) -> gbwtgraph::Payload {
-            gbwtgraph::Payload payload = MIPayload::NO_CODE;
-
-            #pragma omp critical 
-            {
-            //If we've already seen this node before, then return the saved payload
-            if (node_id_to_payload.count(id(pos))) {
-                payload =  node_id_to_payload[id(pos)];
-            }
-            }
-            if (payload != MIPayload::NO_CODE) {
-                return payload;
-            }
-           
+        std::function<gbwtgraph::Payload(const pos_t&)> payload_lambda = 
+            [](const pos_t&) -> gbwtgraph::Payload {
+                return MIPayload::NO_CODE;
+            };
 
-            ZipCode zipcode;
-            zipcode.fill_in_zipcode(*distance_index, pos);
+        gbwtgraph::index_haplotypes(gbz->graph, index, payload_lambda);
+    } else {
+        std::function<gbwtgraph::Payload(const pos_t&)> payload_lambda =
+            [&](const pos_t& pos) -> gbwtgraph::Payload {
+                gbwtgraph::Payload payload = MIPayload::NO_CODE;
 
-            payload = zipcode.get_payload_from_zip();
-            if (payload != MIPayload::NO_CODE) {
-                //If the zipcode is small enough to store in the payload
                 #pragma omp critical 
                 {
-                node_id_to_payload.emplace(id(pos), payload);
+                //If we've already seen this node before, then return the saved payload
+                if (node_id_to_payload.count(id(pos))) {
+                    payload =  node_id_to_payload[id(pos)];
```

Comment:
Looks like the original code was missing a level of indentation here, which probably can be fixed.

Also this should probably be just a `find()` within the critical section, and a comparison to `end()` outside it, so you don't need to `[]` at all and you can do just one lookup.

Though I think this code was here before.